core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.controller;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.JsonFormat;
+import feast.core.config.FeastProperties;
+import feast.core.model.Project;
+import feast.core.service.AccessManagementService;
+import feast.core.service.JobService;
+import feast.core.service.SpecService;
+import feast.core.service.StatsService;
+import feast.proto.core.CoreServiceProto.GetFeastCoreVersionResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureSetRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import feast.proto.core.CoreServiceProto.ListProjectsResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/api/v1", produces = "application/json")
+public class CoreServiceRestController {
+
+  private final JsonFormat.Printer jsonPrinter;
+  private final FeastProperties feastProperties;
+  private SpecService specService;
+  private JobService jobService;
+  private StatsService statsService;
+  private AccessManagementService accessManagementService;
+
+  @Autowired
+  public CoreServiceRestController(
+      FeastProperties feastProperties, SpecService specService,
+      JobService jobService, StatsService statsService,
+      AccessManagementService accessManagementService) {
+    this.feastProperties = feastProperties;
+    this.specService = specService;
+    this.jobService = jobService;
+    this.statsService = statsService;
+    this.accessManagementService = accessManagementService;
+    this.jsonPrinter = JsonFormat.printer();
+  }
+
+  @RequestMapping(value = "/version", method = RequestMethod.GET)
+  @ResponseBody
+  public String getVersion() throws InvalidProtocolBufferException {
+    GetFeastCoreVersionResponse response = GetFeastCoreVersionResponse.newBuilder()
+        .setVersion(feastProperties.getVersion()).build();
+    return jsonPrinter.print(response);
+  }
+
+  @RequestMapping(
+      value = "/project/{project}/feature-set/{featureSetName}",
+      params = {"project", "featureSet"},
```

Comment:
If we're using path variables for this GET request, I don't think we need these params here.

Reply:
can we be consistent with plural/singular?

Reply:
@mrzzy @woop should we use plural or singular here for the path? @mrzzy [suggested singular](https://github.com/feast-dev/feast/pull/878#discussion_r455502972).

Reply:
My preference and what is most commonly followed for collections is plural, although neither is wrong as long as we are consistent.

Reply:
You are being inconsistent by having a `/feature-sets` endpoint and then a different endpoint in `/projects/{project}/feature-set/{featureSetName}`. https://github.com/feast-dev/feast/pull/878/files#diff-8921713d5b60ce56305727b7497dd3d0R130

I think that should be a single path.

Reply:
If we want to be able to list all feature sets across all projects optionally then we should remove `projects` from the hierarchy and make it a query parameter

Reply:
Will be removing this endpoint since can be done in https://github.com/feast-dev/feast/pull/878/files#diff-8921713d5b60ce56305727b7497dd3d0R128.

Reply:
+1 for consistency. Leaning towards `/feature-sets` for list and `/feature-set` for get.